### PR TITLE
{NetAppFiles} Feature netapp add 2023 05 01

### PR DESCRIPTION
**Pull request by audunn**
**Related command**

az netappfiles 

**Description**
Adding support for Azure NetApp Files api-version 2023-05-01.  
Note this migrates the netappfiles module to code generation with customization for backwards compatibility. 

**Testing Guide**

**History Notes**

[NetAppFiles] `az netappfiles check-file-path-availability`: Add new command to check if a file path is available
[NetAppFiles] `az netappfiles check-name-availability`: Add new command to check if a resource name is available
[NetAppFiles] `az netappfiles check-quota-availability`: Add new command to check if a quota is available
[NetAppFiles] `az netappfiles query-network-sibling-set`: Add new command to describe a network sibling set
[NetAppFiles] `az netappfiles update-network-sibling-set`: Add new command to Update the network features of a network sibling set
[NetAppFiles] `az netappfiles quota-limit`: Add new command group to mange quota limits
[NetAppFiles] `az netappfiles volume populate-availability-zone`: Add new command to populate availability zone information for a volume
[NetAppFiles] `az netappfiles volume replication re-initialize`: Add new command to Re-establish a previously deleted replication between 2 volumes that have a common ad-hoc or policy-based snapshots

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
NetAppFiles

File: src/azure-cli/azure/cli/command_modules/network/custom.py [hunk 1]
```diff
@@ -667,7 +667,7 @@ def _build_arguments_schema(cls, *args, **kwargs):
         args_schema.identity = AAZResourceIdArg(
             options=["--identity"],
             help="Name or ID of the ManagedIdentity Resource.",
-            required=True,
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/network/custom.py:670]: Why you made a change in network?

Comment (audunn) [reply src/azure-cli/azure/cli/command_modules/network/custom.py:670]: Good catch. Not sure how that happened.  This should not be there, reverting.

**Comment by audunn** [comment]
Not sure why piprline fails now,   after reverting that change.

File: src/azure-cli/azure/cli/command_modules/netappfiles/_params.py.txt [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long, disable=too-many-statements
+# from azure.cli.core.commands.parameters import tags_type, resource_group_name_type, get_enum_type, get_three_state_flag
+# from azure.mgmt.netapp.models._net_app_management_client_enums import EncryptionKeySource, SmbAccessBasedEnumeration, SmbNonBrowsable, ManagedServiceIdentityType, SecurityStyle
+# from knack.arguments import CLIArgumentType
+
+
+# def load_arguments(self, _):
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/_params.py.txt:12]: Will you plan to remove this file or not?

File: src/azure-cli/azure/cli/command_modules/netappfiles/__init__.py [hunk 3]
```diff
@@ -19,12 +19,23 @@ def __init__(self, cli_ctx=None):
...
+            )
         load_command_table(self, args)
         return self.command_table
 
-    def load_arguments(self, command):
-        from azure.cli.command_modules.netappfiles._params import load_arguments
-        load_arguments(self, command)
+    # def load_arguments(self, command):
+        # from azure.cli.command_modules.netappfiles._params import load_arguments
+        # load_arguments(self, command)
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/__init__.py:38]: actually you should keep this function and an empty _params.py file for furture use.

File: src/azure-cli/azure/cli/command_modules/netappfiles/tests/latest/test_account_commands.py [hunk 4]
```diff
@@ -38,7 +39,7 @@ def test_create_delete_account(self):
         ])
 
         # delete and recheck
-        self.cmd("az netappfiles account delete --resource-group {rg} --account-name '{acc_name}'")
+        self.cmd("az netappfiles account delete --yes --resource-group {rg} --account-name '{acc_name}'")
```
Comment (kairu-ms) [inline src/azure-cli/azure/cli/command_modules/netappfiles/tests/latest/test_account_commands.py:42]: It's a breaking change for the delete command. You may need to remove the confirmation in delete commands currently. You can add that confirmation in the next breaking change window.

**Comment by audunn** [comment]
@kairu-ms  updated

**Comment by kairu-ms** [comment]
Please add tests for the new commands.

**Comment by kairu-ms** [comment]
BTW, your PR removed the command group `az netappfiles account backup`. It's a breaking change. You can do that only in the breaking change window. Do you want to postpone to breaking change release, which is around March 2024.

**Comment by audunn** [comment]
> BTW, your PR removed the command group `az netappfiles account backup`. It's a breaking change. You can do that only in the breaking change window. Do you want to postpone to breaking change release, which is around March 2024.

That is too long for this release.  The issue is a bit tricky since the APIs were removed to be replaced by new APIs in a preview api-version.  Those are supperted in out extension netappfiles-preview.
 
Is there a way to work around this by adding those commands while using the older api-version even if this version is all generated to 2023-05-01?  Perhaps use the old commands and SDK?

**Comment by audunn** [comment]
> BTW, your PR removed the command group `az netappfiles account backup`. It's a breaking change. You can do that only in the breaking change window. Do you want to postpone to breaking change release, which is around March 2024.

@kairu-ms We added support for backup command group using older api version for backwards compatability.

**Comment by audunn** [comment]
@kairu-ms can we move this along now ?

**Comment by jsntcy** [comment]
@audunn,  If you want to release the changes in this PR immediately in a new version, you need update the version and history    notes as well.

**Comment by audunn** [comment]
> @audunn, If you want to release the changes in this PR immediately in a new version, you need update the version and history notes as well.

Hi @jsntcy, history notes are in the pr description, should they be somewhere else as well? Also as this is our first generated version of the main NetApp Files modules I am unsure where do we update the version?

**Comment by audunn** [comment]
@jsntcy can we move this along?

**Comment by necusjz** [comment]
@audunn could you please provide the pr link in aaz repo?

**Comment by audunn** [comment]
> @audunn could you please provide the pr link in aaz repo?

https://github.com/Azure/aaz/pull/424

**Review by necusjz — APPROVED** [review]

**Review by jsntcy — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Comment by audunn** [comment]
@necusjz Do we have an ETA for release of this custmers are waiting for the update?

**Comment by jiasli** [comment]
> [NetAppFiles] `az netappfiles update-network-sibling-set`: Add new command to ~~Update~~update the network features of a network sibling set
> [NetAppFiles] `az netappfiles quota-limit`: Add new command group to ~~mange~~manage quota limits
> [NetAppFiles] `az netappfiles volume replication re-initialize`: Add new command to ~~Re-establish~~re-establish a previously deleted

Please kindly pay attention to spelling and casing, otherwise the history notes (https://github.com/Azure/azure-cli/pull/28642) will need to be fixed manually during releasing.

**Pull request merged**